core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -213,10 +207,10 @@ public void put(Put put) throws CrudException {
         read(key, createGet(key));
       }
       mutationConditionsValidator.checkIfConditionIsSatisfied(
-          put, snapshot.getFromReadSet(key).orElse(null));
+          put, snapshot.getResult(key).orElse(null));
```

Comment:
Use the added `getResult()` method to use the result that's merged with a mutation for the same record in the write set for conditional updates.